core/org.eclipse.cdt.ui/src/org/eclipse/cdt/ui/newui/AbstractCPropertyTab.java

```diff
@@ -576,8 +585,14 @@ public IStatus validate(Object[] selection) {
 		if (dialog.open() == Window.OK) {
 			IResource resource = (IResource) dialog.getFirstResult();
 			if (resource != null) {
+				if (resource.getProject() == prj) {
```

Comment:
I think this should use .equals? ref: https://github.com/eclipse-cdt/cdt/blob/7c8bb9f00ef0d5e3ff493bc0c944519a3e476da2/core/org.eclipse.cdt.ui/utils.ui/org/eclipse/cdt/utils/ui/controls/FileListControl.java#L325

```suggestion
				if (resource.getProject().equals(prj)) {
```

Reply:
I wasn't aware of the equals. Thanks.